packages/relay/src/lib/eth.ts

```diff
@@ -580,21 +582,20 @@ export class EthImpl implements Eth {
       }
     }
 
-    // check cache first
-    // create a key for the cache
-    const cacheKey = `${constants.CACHE_KEY.ETH_GET_BALANCE}-${account}-${blockNumberOrTag}`;
-    const cachedBalance = this.cache.get(cacheKey);
-    if (cachedBalance) {
-      this.logger.trace(`${requestIdPrefix} returning cached value ${cacheKey}:${JSON.stringify(cachedBalance)}`);
-      return cachedBalance;
-    }
-
     let blockNumber = null;
     let balanceFound = false;
     let weibars: BigInt = BigInt(0);
     const mirrorAccount = await this.mirrorNodeClient.getAccount(account, requestId);
 
     try {
+      // check balance cache first for high load scenarios
+      const cacheKey = `${constants.CACHE_KEY.ETH_GET_BALANCE}-${account}-${blockNumberOrTag}`;
+      const cachedBalance = this.cache.get(cacheKey);
+      if (cachedBalance) {
+        this.logger.trace(`${requestIdPrefix} returning cached value ${cacheKey}:${JSON.stringify(cachedBalance)}`);
+        return cachedBalance;
+      }
+
       if (!EthImpl.blockTagIsLatestOrPending(blockNumberOrTag)) {
         const block = await this.getHistoricalBlockResponse(blockNumberOrTag, true, requestId);
```

Comment:
1. line 600 - this is the first call that will call the blocks API with `blockNumberOrTag`
2. line 606 - we call block for latest just so we can check if it's not the same as `blockNumberOrTag`. I forget why we need the latest at all. Don't we just want to know the balance at `blockNumberOrTag`. One consideration is if `blockNumberOrTag` is not latest or pending or earliest then we know its a valid number. In which case we should just use the block timestamp and call the balances API for the account and get that timestamp. Use that timestamp and compare it's block number to `blockNumberOrTag`. 

If the diff is over 2 sec and under 15 mins then do the calculation of balance by getting the crypto transfers between balance timestamp and the block time. Otherwise return that balance

Reply:
Current thinking but let's see

Reply:
Agreed and as discussed, this will be address in a following issue and PR.